controllers/idler/idler_controller_test.go

```diff
@@ -594,6 +594,7 @@ func TestCreateNotification(t *testing.T) {
 		},
 		Spec: toolchainv1alpha1.IdlerSpec{TimeoutSeconds: 60},
 	}
+
```

Comment:
The current tests are not enough to cover the new functionality.
I would add a test add use the `prepareReconcile()` function which already creates a tree of different workloads with different controllers. Then call the `scaleControllerToZero()` function for each type/case and check the outcome AppName/Type.